docs/proposals/PROPOSAL_F.md

```diff
@@ -0,0 +1,374 @@
+         }
+      },
+      {
+         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+         "digest": "sha256:4ff3ca91275773af45cb4b0834e12b7eb47d1c18f770a0b151381cd227f4c253",
+         "size": 528,
+         "platform": {
+            "architecture": "amd64",
+            "os": "linux"
+         },
+         // signatures added to the linux/amd64
+         "annotations": {
+            "dev.cosignproject.cosign/signature.e59879c": "...",
+            "dev.sigstore.cosign/bundle.6436749": "...",
+            "dev.sigstore.cosign/certificate.776a3f1": "...",
+            "dev.sigstore.cosign/chain.8adf60e": "...",
+            "dev.cosignproject.cosign/signature.de25bf6": "...",
+            "dev.sigstore.cosign/bundle.37d34ff": "...",
+            "dev.sigstore.cosign/certificate.e8adf60": "...",
+            "dev.sigstore.cosign/chain.b1c64a3": "...",
+            "org.opencontainers.signatures.index":"dev.cosignproject.cosign/signature.e59879c,dev.cosignproject.cosign/signature.de25bf6,dev.sigstore.cosign/bundle.37d34ff,dev.sigstore.cosign/certificate.e8adf60,dev.sigstore.cosign/chain.b1c64a3,dev.sigstore.cosign/bundle.6436749,dev.sigstore.cosign/certificate.776a3f1,dev.sigstore.cosign/chain.8adf60e"
+         }
+      },
+      // artifacts referencing the image manifests
+      {
+         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+         "digest": "sha256:5b0044a1244...", // arm64 SBOM
+         "size": 1024,
+         "annotations": {
+            "org.opencontainers.reference.type": "sbom",
+            "org.opencontainers.reference.digest": "sha256:c3c58223e2af75154c4a7852d6924b4cc51a00c821553bbd9b3319481131b2e0",
+            "org.opencontainers.reference.description": "Software Bill of Materials of alpine:3.17 linux/arm64 image"
+         }
+      },
+      {
+         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+         "digest": "sha256:5b0044a44d...", // amd64 SBOM
+         "size": 1024,
+         "annotations": {
+            "org.opencontainers.reference.type": "sbom",
+            "org.opencontainers.reference.digest": "sha256:4ff3ca91275773af45cb4b0834e12b7eb47d1c18f770a0b151381cd227f4c253",
+            "org.opencontainers.reference.description": "Software Bill of Materials of alpine:3.17 linux/amd64 image"
+         }
+      }
+   ]
+}
+```
+
+### Registry HTTP API
+
+> The distribution-spec **SHOULD** solve the race condition occurring when [multiple parties modify an OCI Index in parallel](https://github.com/opencontainers/distribution-spec/pull/251).
+Querying for a reference involves pulling either existing tags pointing to an index that includes the artifact and image, or pulling new tags that use the digest of the referenced image in the tag value.
+
+> Registries are expected to handle nested OCI Index correctly.
```

Comment:
This will need testing.

Reply:
Added `Testing is needed to verify that a registry supports nested OCI Index.` in the `Backwards Compatibility.2` section.

Reply:
Fixed https://github.com/opencontainers/wg-reference-types/pull/52/commits/47a0daa6a00ac31c25ba0d180c158ff9442e6634

Reply:
@silvin-lubecki - What is the recommendation when the index size is more than 4MB. Implementations currently reject manifests that are too large.

Reply:
That was discussed in https://github.com/opencontainers/distribution-spec/issues/260 and officially clarified in https://github.com/opencontainers/distribution-spec/pull/293 :sweat_smile: